core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/arrayfunction/ArrayContainsExpression.kt

```diff
@@ -1,15 +1,28 @@
 package ch.ergon.dope.resolvable.expression.unaliased.type.arrayfunction
 
+import ch.ergon.dope.DopeQuery
 import ch.ergon.dope.resolvable.expression.TypeExpression
 import ch.ergon.dope.resolvable.expression.unaliased.type.toDopeType
+import ch.ergon.dope.resolvable.operator.FunctionOperator
 import ch.ergon.dope.validtype.ArrayType
 import ch.ergon.dope.validtype.BooleanType
 import ch.ergon.dope.validtype.NumberType
 import ch.ergon.dope.validtype.StringType
 import ch.ergon.dope.validtype.ValidType
 
-class ArrayContainsExpression<T : ValidType>(array: TypeExpression<ArrayType<T>>, value: TypeExpression<T>) :
-    ArrayFunctionNumberExpression<T>("ARRAY_CONTAINS", array, value)
+class ArrayContainsExpression<T : ValidType>(
+    private val array: TypeExpression<ArrayType<T>>,
+    private val value: TypeExpression<T>,
+) : TypeExpression<BooleanType>, FunctionOperator {
+    override fun toDopeQuery(): DopeQuery {
+        val arrayDopeQuery = array.toDopeQuery()
+        val valueDopeQuery = value.toDopeQuery()
+        return DopeQuery(
+            queryString = toFunctionQueryString("ARRAY_CONTAINS", arrayDopeQuery, valueDopeQuery),
+            parameters = arrayDopeQuery.parameters + valueDopeQuery.parameters,
+        )
+    }
+}
 
 fun <T : ValidType> arrayContains(array: TypeExpression<ArrayType<T>>, value: TypeExpression<T>) =
```

Comment:
I think these functions should all be extensiosn-functions instead, something like:
`fun <T : ValidType> TypeExpression<ArrayType<T>>.contains(value: TypeExpression<T>) = ArrayContainsExpression(this, value)`
(same for all the other functions)
You should be able to write something like
```
.select(someStringArrayField().prepend("lol").min())
.where(someIntArrayField().concat(someIntArrayField()).contains(12))
```
But maybe you can do that in another PR

Reply:
I am going to do that in another PR, as to not inflate this PR further.